src/LibraryManager/Providers/Cdnjs/CdnjsProvider.cs

```diff
@@ -237,35 +239,21 @@ public string GetSuggestedDestination(ILibrary library)
 
         private async Task<Stream> GetStreamAsync(ILibraryInstallationState state, string sourceFile, CancellationToken cancellationToken)
         {
-            string name = GetLibraryName(state.LibraryId);
-            string version = GetLibraryVersion(state.LibraryId);
+            (string name, string version) = LibraryNamingScheme.Instance.GetLibraryNameAndVersion(state.LibraryId);
 
-            string absolute = Path.Combine(CacheFolder, name, version, sourceFile);
-
-            if (File.Exists(absolute))
+            if (!string.IsNullOrEmpty(name) && !string.IsNullOrEmpty(version))
             {
-                return await HostInteraction.ReadFileAsync(absolute, cancellationToken).ConfigureAwait(false);
+                string absolute = Path.Combine(CacheFolder, name, version, sourceFile);
+
+                if (File.Exists(absolute))
```

Comment:
>File [](start = 20, length = 4)

Can you see if we hit this on the UI thread (e.g. with a UNC path)?

Reply:
The path is always a local path to the local cache folder

---
In reply to: [197970287](https://github.com/aspnet/LibraryManager/pull/230#discussion_r197970287) [](ancestors = 197970287)